packages/next-app/src/components/Sidebar.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+
+import { MagnifyingGlassIcon } from "@radix-ui/react-icons";
+import classNames from "classnames";
+import { createContext, useContext } from "react";
+
+interface SidebarContextType {
+  isFloating: boolean;
+}
+
+const SidebarContext = createContext({} as SidebarContextType);
+
+function useSidebarContext() {
+  const context = useContext(SidebarContext);
+
+  if (!context) {
+    throw new Error(
+      "Child components of Sidebar cannot be rendered outside the Sidebar component!"
+    );
+  }
+
+  return context;
+}
+
+export default function Sidebar({
+  children,
+  isFloating = false,
+}: {
+  children: React.ReactNode;
+  isFloating?: boolean;
+}) {
+  return (
+    <SidebarContext.Provider value={{ isFloating }}>
+      <div
+        className={classNames(
+          "w-full max-w-full  py-5",
+          isFloating ? "bg-gray-900" : "bg-gray-800"
+        )}
+      >
+        <div className="h-screen w-96 max-w-full items-start justify-start space-y-4">
+          {children}
+        </div>
+      </div>
+    </SidebarContext.Provider>
+  );
+}
+
+function Header({
+  children,
+  name,
+}: {
+  children?: React.ReactNode;
+  name: string;
+}) {
+  useSidebarContext();
```

Comment:
❓

Reply:
I thought it was overcomplicating doing a new component just for the Header Title which has this code: 

```
<div className="flex items-center justify-start space-x-0 text-2xl font-medium text-gray-400">
    <span>{name}</span>
</div>
```

And I used the name as a props. But now thinking about it the user might not want to have a Header title, and with this I'm forcing it...

My new approach would be:

```
function Header({
  children
}: {
React.PropsWithChildren) {
  useSidebarContext();
  return (
    <div className="items-start justify-start space-y-2.5 self-stretch px-5">
      {children}
    </div>
  );
}

function HeaderTitle({
children
}:{React.PropsWithChildren}) {
useSidebarContext();
return(
  <div className="flex items-center justify-start space-x-0 text-2xl font-medium text-gray-400">
    <span>{name}</span>
  </div>
)}
```

WDYT?

Reply:
I think that makes sense -- I also like this approach better than the first one!